include/drivers/interrupt_controller/gic.h

```diff
@@ -219,6 +210,115 @@
 
 #endif /* CONFIG_GIC_VER <= 2 */
 
+/* GICC_IAR */
+#define GICC_IAR_SPURIOUS		1023
+
+/* GICD_ICFGR */
+#define GICD_ICFGR_MASK			BIT_MASK(2)
+#define GICD_ICFGR_TYPE			BIT(1)
+
+/* GICD_TYPER.ITLinesNumber 0:4 */
+#define IT_LINES_NO_MASK		0x1f
+
+/*
+ * Helper Constants
+ */
+
+#define	GIC_SPI_INT_BASE		32
+
+#define NUM_INTR_PER_REG		32
+
+#define NUM_CFG_PER_REG			16
+
+#define NUM_PRI_PER_REG			4
+
+/* GIC idle priority : value '0xff' will allow all interrupts */
+#define GIC_IDLE_PRIO			0xff
+/* Priority levels 0:255 */
+#define GIC_PRI_MASK			0xff
+
+/*
+ * '0xa0'is used to initialize each interrtupt default priority.
+ * This is an arbitrary value in current context.
+ * Any value '0x80' to '0xff' will work for both NS and S state.
+ * The values of individual interrupt and default has to be chosen
+ * carefully if PMR and BPR based nesting and preemption has to be done.
+ */
+#define GIC_INT_DEF_PRI_X4		0xa0a0a0a0
+
+#if defined(CONFIG_GIC_V3)
+
+#define GIC_RDIST_BASE		DT_REG_ADDR_BY_IDX(DT_INST(0, arm_gic), 1)
+
+/* Fixme: update from platform define or dt */
+#define GIC_NUM_CPU_IF			1
+/* Fixme: get current core id from mpidr */
+#define GET_CPUID			0
+
+#define GIC_GET_RDIST(cpuid)		gic_rdists[cpuid]
+
+/* SGI base is at 64K offset from Redistributor */
+#define GICR_SGI_BASE_OFF		0x10000
+
+/* GICR registers offset from RD_base(n) */
+#define GICR_CTLR			0x0000
+#define	GICR_IIDR			0x0004
+#define	GICR_TYPER			0x0008
+#define GICR_STATUSR			0x0010
+#define GICR_WAKER			0x0014
```

Comment:
Nits: I know this is not entirely your fault, but can we stick with `#define<SPACE>` (instead of `#define<TAB>`) as with the rest?

Reply:
ACK. sorry I missed.